src/api/app/controllers/webui/package_controller.rb

```diff
@@ -984,18 +984,40 @@ def meta
   end
 
   def save_meta
-    begin
-      frontend.put_file(params[:meta], :project => @project, :package => @package, :filename => '_meta')
-    rescue ActiveXML::Transport::Error => e
-      message = e.summary
-      flash[:error] = message
-      @meta = params[:meta]
-      render :text => message, :status => 400, :content_type => 'text/plain'
+    meta_xml = Xmlhash.parse(params[:meta])
+    unless meta_xml
+      render(text: "Invalid XML file", content_type: 'text/plain', status: 400)
```

Comment:
I'd also do
``
flash.now[:success] = message
render layout: false, partial: 'layouts/webui/flash', object: flash
``
because then we show a proper flash message and the user know what goes wrong instead of just show a red button!